lib/middleware.ts

```diff
@@ -52,7 +52,7 @@ export function applyMiddleware (
             Object.assign(initialObj, middlewareObj)
           )
       } else {
-        const result = middleware(argv, yargs)
+        const result = middleware(accumulation, yargs)
```

Comment:
Although I promised not to mix switching to ts and refactoring... done :-)

`accumulation` was the actual name of the `reduce` parameter, so that is why I kept it.